wunderio/next-drupal-starterkit
# NEX-143: Add pRetry to calls from frontend to Drupal

## Link to ticket:

https://wunder.atlassian.net/browse/NEX-143

## Link to feature environment:

*  https://nex-143-next.next-drupal-starterkit.dev.wdr.io/
*  https://nex-143.next-drupal-starterkit.dev.wdr.io


## Changes proposed in this PR:

Starting from a shared experience on the drupal next.js slack (https://drupal.slack.com/archives/C01E36BMU72/p1708638462166279?thread_ts=1708638293.534419&cid=C01E36BMU72) 

![image](https://github.com/wunderio/next-drupal-starterkit/assets/185412/f9043486-218b-4a05-97a1-53433ceb9f7a)


We are adding this npm package: 

https://www.npmjs.com/package/p-retry

And wrapping our calls to drupal with it, configuring it to retry 5 times in case of errors.

This should make the builds more resilient, expecially in the dev cluster in silta we have seen relatively often failed builds.

## How to test:

We should see fewer failures in builds (And during site operations), since the frontend should be able to retry to call Drupal if a request does not go well.
